models.py

```diff
@@ -0,0 +1,53 @@
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
+from sqlalchemy import Integer, Numeric, String, ARRAY, Date, Time, ForeignKey, func
+from config import engine
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class Asset(Base):
+    __tablename__ = "assets"
+    asset_id: Mapped[int] = mapped_column(Integer, primary_key=True)
+    asset_name: Mapped[str] = mapped_column(String(50), unique=True, nullable=False)
+    asset_ticker: Mapped[str] = mapped_column(String(10), nullable=False)
+    asset_data: Mapped[list["AssetData"]] = relationship("AssetData", back_populates="asset", cascade="all, delete-orphan")
+
+
+class AssetData(Base):
+    __tablename__ = "asset_data"
+    data_id: Mapped[int] = mapped_column(Integer, primary_key=True)
+    asset_id: Mapped[int] = mapped_column(ForeignKey("assets.asset_id", ondelete="CASCADE"))
```

Comment:
```suggestion
    asset_id: Mapped[int] = mapped_column(ForeignKey("assets.asset_id", ondelete="CASCADE"), index=True)
```

For any columns you want to apply filters to, you want to add an index to make the look up log(n)